Descrption: No gksu/disable-grab option per gui in gnome-at-properties
Author: ?

diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.91/capplets/accessibility/at-properties/at-enable-dialog.ui gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/at-enable-dialog.ui
--- gnome-control-center-2.27.91/capplets/accessibility/at-properties/at-enable-dialog.ui	2009-08-31 02:16:12.377854799 +0100
+++ gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/at-enable-dialog.ui	2009-08-31 02:16:15.337854165 +0100
@@ -98,6 +98,28 @@
                                 <property name="position">1</property>
                               </packing>
                             </child>
+                            <child>
+                              <object class="GtkCheckButton" id="at_gksudo_toggle">
+                                <property name="visible">True</property>
+                                <property name="can_focus">True</property>
+                                <property name="label" translatable="yes">_Password dialogs as normal windows</property>
+                                <property name="use_underline">True</property>
+                                <property name="draw_indicator">True</property>
+                                <accessibility>
+                                  
+                                </accessibility>
+                                <child internal-child="accessible">
+                                  <object class="AtkObject" id="a11y-at_gksudo_toggle1">
+                                    <property name="AtkObject::accessible_description" translatable="yes">Changes to enable assistive technologies will not take effect until your next log in.</property>
+                                  </object>
+                                </child>
+                              </object>
+                              <packing>
+                                <property name="expand">False</property>
+                                <property name="fill">False</property>
+                                <property name="position">2</property>
+                              </packing>
+                            </child>
                           </object>
                           <packing>
                             <property name="position">1</property>
diff -Nur -x '*.orig' -x '*~' gnome-control-center-2.27.91/capplets/accessibility/at-properties/main.c gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/main.c
--- gnome-control-center-2.27.91/capplets/accessibility/at-properties/main.c	2009-08-31 02:16:12.373856697 +0100
+++ gnome-control-center-2.27.91.new/capplets/accessibility/at-properties/main.c	2009-08-31 02:16:15.337854165 +0100
@@ -21,6 +21,8 @@
 
 #define ACCESSIBILITY_KEY       "/desktop/gnome/interface/accessibility"
 #define ACCESSIBILITY_KEY_DIR   "/desktop/gnome/interface"
+#define GKSU_NO_GRAB_KEY        "/apps/gksu/disable-grab"
+#define GKSU_NO_GRAB_KEY_DIR    "/apps/gksu"
 
 static gboolean initial_state;
 
@@ -211,6 +213,42 @@
 }
 
 static void
+at_gksudo_update  (GConfClient *client,
+                   GtkBuilder  *builder)
+{
+       gboolean is_enabled = gconf_client_get_bool (client, GKSU_NO_GRAB_KEY, NULL);
+       GObject *button = gtk_builder_get_object (builder, "at_gksudo_toggle");
+       
+       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (button),
+                                     is_enabled);
+}
+
+
+static void
+at_gksudo_changed (GConfClient *client,
+                  guint        cnxn_id,
+                  GConfEntry  *entry,
+                  gpointer     user_data)
+{
+       at_gksudo_update (client, user_data); 
+}
+
+
+static void
+at_gksudo_toggled (GtkToggleButton *toggle_button, 
+                   GtkBuilder      *builder)
+{
+       GConfClient *client = gconf_client_get_default ();
+       gboolean is_enabled = gtk_toggle_button_get_active (toggle_button);
+       
+       gconf_client_set_bool (client, GKSU_NO_GRAB_KEY,
+                              is_enabled,
+                              NULL);
+       g_object_unref (client);
+}
+
+
+static void
 setup_dialog (GtkBuilder *builder)
 {
 	GConfClient *client;
@@ -219,7 +257,27 @@
 	GObject *peditor;
 
 	client = gconf_client_get_default ();
-
+        
+    //Password grab toggle
+    gconf_client_add_dir (client, GKSU_NO_GRAB_KEY_DIR, 
+                          GCONF_CLIENT_PRELOAD_ONELEVEL, NULL);
+    
+    object = gtk_builder_get_object (builder, "at_gksudo_toggle");
+        
+    g_signal_connect (object, "toggled",
+                      G_CALLBACK (at_gksudo_toggled),   
+                      builder);
+
+    peditor = gconf_peditor_new_boolean (NULL, GKSU_NO_GRAB_KEY,
+                                         GTK_WIDGET (object),
+                                         NULL);     
+        
+    at_gksudo_update (client, builder);
+       
+    gconf_client_notify_add (client, GKSU_NO_GRAB_KEY_DIR,
+        at_gksudo_changed,
+        builder, NULL, NULL);
+            
 	gconf_client_add_dir (client, ACCESSIBILITY_KEY_DIR,
 			      GCONF_CLIENT_PRELOAD_ONELEVEL, NULL);
 
